state-sync/state-sync-v2/state-sync-driver/src/driver.rs

```diff
@@ -139,6 +150,45 @@ impl<
     async fn handle_consensus_notification(&mut self, notification: ConsensusNotification) {
         debug!("Received a consensus notification: {:?}", notification);
 
+        // Verify the notification: full nodes shouldn't receive notifications
+        // and consensus should only send notifications after bootstrapping!
+        let result = if self.driver_configuration.role == RoleType::FullNode {
+            Err(Error::FullNodeConsensusNotification(format!(
+                "Received a consensus notification: {:?}",
+                notification
+            )))
+        } else if !self.bootstrapper.is_bootstrapped() {
+            Err(Error::BootstrapNotComplete(format!(
+                "Consensus shouldn't be running! Unable to process notification: {:?}",
+                notification
+            )))
```

Comment:
Message is a little confusing saying consensus shouldn't be running, making it seem like it's an issue because consensus is ahead.  I think it should probably focus on the fact that state sync isn't ready.  Maybe say something around "Received consensus message, but not yet bootstrapped"?

Reply:
Done, updated this to "Received consensus notification: {:?}, notification". The error wrapper `BootstrapNotComplete` will mention that state sync hasn't finished bootstrapping.